independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/PathTreeWithManifest.java

```diff
@@ -0,0 +1,164 @@
+package io.quarkus.paths;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.jar.Manifest;
+import java.util.stream.Stream;
+import org.jboss.logging.Logger;
+
+public abstract class PathTreeWithManifest implements PathTree {
+
+    private static final String META_INF_VERSIONS = "META-INF/versions/";
+    public static final int JAVA_VERSION;
+
+    static {
+        int version = 8;
+        try {
+            Method versionMethod = Runtime.class.getMethod("version");
+            Object v = versionMethod.invoke(null);
+            List<Integer> list = (List<Integer>) v.getClass().getMethod("version").invoke(v);
+            version = list.get(0);
+        } catch (Exception e) {
+            //version 8
```

Comment:
Do we even support Java 8 anymore? pom.xml says no if I'm not mistaken.

Reply:
We don't. I copied this code from another class, tbh. I'll change it to fail hard.